src/main/java/frc/trigon/robot/commands/commandfactories/CollectionCommands.java

```diff
@@ -0,0 +1,62 @@
+import frc.trigon.robot.commands.CommandConstants;
+import frc.trigon.robot.commands.commandclasses.CoralAlignmentCommand;
+import frc.trigon.robot.subsystems.algaeintake.AlgaeIntakeCommands;
+import frc.trigon.robot.subsystems.algaeintake.AlgaeIntakeConstants;
+import frc.trigon.robot.subsystems.coralintake.CoralIntakeCommands;
+import frc.trigon.robot.subsystems.coralintake.CoralIntakeConstants;
+import frc.trigon.robot.subsystems.gripper.GripperCommands;
+import frc.trigon.robot.subsystems.gripper.GripperConstants;
+import org.trigon.hardware.misc.leds.LEDCommands;
+
+/**
+ * A class that contains commands for collecting game pieces.
+ */
+public class CollectionCommands {
+    public static boolean SHOULD_ALIGN_TO_CORAL = true;
+
+    public static Command getAlgaeCollectionCommand() {
+        return new ParallelCommandGroup(
+                AlgaeIntakeCommands.getSetTargetStateCommand(AlgaeIntakeConstants.AlgaeIntakeState.COLLECT),
+                LEDCommands.getBlinkingCommand(Color.kAqua, AlgaeIntakeConstants.COLLECTION_LEDS_BLINKING_SPEED)
+        );
+    }
+
+    public static Command getCoralCollectionCommand() {
+        return getInitiateCollectionCommand()
+                .unless(RobotContainer.CORAL_INTAKE::hasGamePiece)
+                .alongWith(CommandConstants.COLLECTION_RUMBLE_COMMAND)
+                .onlyIf(RobotContainer.CORAL_INTAKE::hasGamePiece);
+    }
+
+    private static Command getInitiateCollectionCommand() {
+        return new ParallelCommandGroup(
+                new CoralAlignmentCommand().onlyIf(() -> SHOULD_ALIGN_TO_CORAL),
+                LEDCommands.getBlinkingCommand(Color.kAqua, CoralIntakeConstants.COLLECTION_LEDS_BLINKING_SPEED).unless(() -> SHOULD_ALIGN_TO_CORAL),
+                CoralIntakeCommands.getSetTargetStateCommand(CoralIntakeConstants.CoralIntakeState.COLLECT),
+                GripperCommands.getSetTargetStateCommand(GripperConstants.GripperState.LOAD_CORAL)
+        ).until(RobotContainer.CORAL_INTAKE::isEarlyCoralCollectionDetected).finallyDo(() -> getRetractCoralIntakeAndFeedToGripperCommand().schedule());
+    }
+
+    private static Command getRetractCoralIntakeAndFeedToGripperCommand() {
+        return new SequentialCommandGroup(
+                getRetractCoralIntakeCommand(),
+                GeneralCommands.runWhen(CoralIntakeCommands.getSetTargetStateCommand(CoralIntakeConstants.CoralIntakeState.FEED), RobotContainer.CORAL_INTAKE::atTargetAngle),
+                GeneralCommands.runWhen(CoralIntakeCommands.getStopCommand(), RobotContainer.GRIPPER::hasGamePiece)
+        );
+    }
+
+    private static Command getRetractCoralIntakeCommand() {
+        return GeneralCommands.getContinuousConditionalCommand(
+                CoralIntakeCommands.getSetTargetStateCommand(CoralIntakeConstants.CoralIntakeState.RETRACT).until(() -> RobotContainer.CORAL_INTAKE.atState(CoralIntakeConstants.CoralIntakeState.RETRACT)),
+                CoralIntakeCommands.getStopCommand(),
```

Comment:
You don't want to stop going to the retracted position once you detect a game piece. You want to stop moving the motors collection motors but keep moving to the target position